src/extract-links/from-dom.ts

```diff
@@ -67,34 +92,41 @@ function linksInAttribute({ element, attributeInfo, baseUrl, docUrl }) {
                 get element() { return element },
                 get attribute() { return attribute },
                 get rangeWithinAttribute() {
-                    return [index, index + tokenView.token.length]
+                    return [index, index + tokenView.token.length] as [number, number]
                 },
             }
         },
 
         // These values are constant, but we use getters anyway to emphasise they are read-only.
         get isSubresource() { return attributeInfo.isSubresource },
         get subresourceType() { return attributeInfo.subresourceType },
-    }))
+    } as HtmlLink)) // TODO actually typecheck; may need a revisit of attribute info format.
```

Comment:
This makes me a little unhappy, but I did not see an easy way to map between the known attributes listed in `src/extract-links/url-attributes/attribute-lists.ts` and the link types in `extract-links/types.ts`. Perhaps inspiration will come another day.